src/reporter.js

```diff
@@ -199,24 +201,42 @@ function CoverageIstanbulReporter(baseReporterDecorator, logger, config) {
     }
   }
 
+  function writeReports(config, coverageMap, sourceFinder) {
+    const dir = path.resolve(config.reporting.dir);
+    const contextOpts = {
+      dir,
+      watermarks: config.reporting.watermarks || [],
+      sourceFinder,
+      coverageMap
+    };
+    const context = libReport.createContext(contextOpts);
+
+    const tree = context.getTree(config.summarizer);
+    config.reporting.reports.forEach(name => {
+      const reportConfig = (config.reporting['report-config'] || [])[name];
+      const report = libReports.create(name, reportConfig);
+      tree.visit(report, context);
+    });
+  }
+
   this.onBrowserComplete = function(browser, result) {
     if (result && result.coverage) {
       browserCoverage.set(browser, result.coverage);
     }
   };
 
   const baseReporterOnRunComplete = this.onRunComplete;
-  this.onRunComplete = function(browsers, results) {
-    // eslint-disable-next-line prefer-rest-params
-    baseReporterOnRunComplete.apply(this, arguments);
-
+  this.onRunComplete = async function(browsers, results) {
     if (coverageConfig.combineBrowserReports) {
-      createReport(browsers, results);
+      await createReport(browsers, results);
     } else {
-      browsers.forEach(browser => {
+      await browsers.forEach(browser => {
```

Comment:
You can only await a promise, so this needs to be something like `await Promise.all(browsers.map(browser => createReport(browser, results)))`

Reply:
Thanks for catching that. Fixed it.